Cal log details

Shanefe opened the pull request:
Add isOrganizer and IsRoom, Cleanup unreliable columns

dpaulson45 commented on Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:
```diff
@@ -188,6 +186,7 @@ function GetCalendarDiagnosticObjects {
     }
 
     Write-Host "Found $($CalLogs.count) Calendar Logs for [$Identity]"
+    $script:IsOrganizer = SetIsOrganizer($CalLogs)
```
I am not seeing where this variable is being used `$script:IsOrganizer`

Shanefe replied:
Now on line 1610.

dpaulson45 replied:
This isn't being used still as you are resetting the value to `$IsOrganizer`. 

https://github.com/microsoft/CSS-Exchange/pull/1979/files#diff-c0a17053cc97c9900333ba5ee8350f54a57850310cf6cb5bf2ed9e64fec2de2aR1611

Shanefe replied:
Removed!

dpaulson45 commented on Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:
```diff
@@ -257,6 +256,46 @@ function GetMailbox {
+    return $IsOrganizer
+}
+
+function SetIsRoom {
```
function is created, but never used.

Shanefe replied:
Both of these functions are for a new feature that Rui is adding, direct export to Excel using OpenXML with formatting based of if they are the Organizer or if this is a Room MB supplied with these functions.

dpaulson45 replied:
Okay, if it isn't going to be used till a later date with some other changes, it should be checked in at that time so everything can be reviewed all at once and have a better understanding of what is occurring.

Shanefe replied:
Now is on Line 1612.

dpaulson45 commented on Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:
```diff
@@ -308,7 +347,7 @@ function GetMailboxProp {
     )
 
-    Write-Verbose "GetMailboxProp: [$Prop]: Searching for:[$PassedCN]..."
+    Write-Debug "GetMailboxProp: [$Prop]: Searching for:[$PassedCN]..."
```
Do you want to keep this as a Debug?

Shanefe replied:
Yes, way too much information on Verbose.

Shanefe commented:
I believe that I am done with the latest round of changes.

Wish I had got the PS Testing done.
Looks like we will be moving a lot of this logic out of PS and into a cmdlet, so the script should get a lot simpler in the future.

Let me know if you have any requests on changing the code, etc.

Shane

From: David Paulson ***@***.***>
Sent: Friday, February 23, 2024 10:30 AM
To: microsoft/CSS-Exchange ***@***.***>
Cc: Shane Ferrell ***@***.***>; Author ***@***.***>
Subject: Re: [microsoft/CSS-Exchange] Cal log details (PR #1979)


@dpaulson45 commented on this pull request.

________________________________

In Calendar/Get-CalendarDiagnosticObjectsSummary.ps1<https://github.com/microsoft/CSS-Exchange/pull/1979#discussion_r1501032128>:

> @@ -188,6 +186,7 @@ function GetCalendarDiagnosticObjects {

     }



     Write-Host "Found $($CalLogs.count) Calendar Logs for [$Identity]"

+    $script:IsOrganizer = SetIsOrganizer($CalLogs)

This isn't being used still as you are resetting the value to $IsOrganizer.

https://github.com/microsoft/CSS-Exchange/pull/1979/files#diff-c0a17053cc97c9900333ba5ee8350f54a57850310cf6cb5bf2ed9e64fec2de2aR1611

-
Reply to this email directly, view it on GitHub<https://github.com/microsoft/CSS-Exchange/pull/1979#discussion_r1501032128>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABPSHT5DAKYY2RV6JPXPIQ3YVDN3DAVCNFSM6AAAAABDALMRT2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQOJYG4YDOMBXGA>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

dpaulson45 commented:
I will try to review it this week.

dpaulson45 commented on Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:
```diff
@@ -257,6 +256,66 @@ function GetMailbox {
+        if ($CalendarItemTypes.($CalLog.ItemClass) -eq "IpmAppointment" -and
+            $CalLog.ExternalSharingMasterId -eq "NotFound" -and
+            ($CalLog.ResponseType -eq "1" -or $CalLogs.ResponseType -eq "Organizer")) {
+            $IsOrganizer = "True"
```
This should be a boolean of `$true`, not a string if we are originally setting `$IsOrganizer = $false`

Shanefe replied:
Done

dpaulson45 commented on Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:
```diff
@@ -257,6 +256,66 @@ function GetMailbox {
+        if ($CalendarItemTypes.($CalLog.ItemClass) -eq "IpmAppointment" -and
+            $CalLog.ExternalSharingMasterId -eq "NotFound" -and
+            $CalLog.Client -eq "ResourceBookingAssistant" ) {
+            $IsRoom = "True"
```
Same thing here. Let's be consistent, either a string or a boolean

dpaulson45 commented on Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:
```diff
@@ -257,6 +256,66 @@ function GetMailbox {
+        if ($CalendarItemTypes.($CalLog.ItemClass) -eq "IpmAppointment" -and
+            $CalLog.ExternalSharingMasterId -eq "NotFound" -and
+            $CalLog.CalendarItemType.ToString() -eq "RecurringMaster" ) {
+            $IsRecurring = "True"
```
Same thing here. Let's be consistent, either a string or a boolean

Shanefe replied:
Done

dpaulson45 commented on Calendar/Get-CalendarDiagnosticObjectsSummary.ps1:
```diff
@@ -374,13 +433,21 @@ function BetterThanNothingCNConversion {
         if ($cNameMatch[-1] -match "-[\w* -.]*") {
-            Write-Verbose "BetterThanNothingCNConversion: Returning : [$($cNameMatch[-1])]"
-            return $cNameMatch.split('-')[-1]
+            Write-Verbose "BetterThanNothingCNConversion: Matched : [$($cNameMatch[-1])]"
```
Wouldn't it be better if you just do the last X number of the CN? If we don't want it to be so long, that might be an idea.

Shanefe replied:
Unfortunately, there is enough variation in the CN formation, that the last X often is worse than this convoluted system that we have. This is likely getting moved into the Dev code in a few months, so it will be removed / clean up from here.

Shanefe commented:
Thanks for the PR review.  I think I have addressed all comments.
Please let me know if you have any other comments / etc. before this can go in.

Shane
________________________________
From: David Paulson ***@***.***>
Sent: Wednesday, March 20, 2024 6:47 AM
To: microsoft/CSS-Exchange ***@***.***>
Cc: Shane Ferrell ***@***.***>; Author ***@***.***>
Subject: Re: [microsoft/CSS-Exchange] Cal log details (PR #1979)


@dpaulson45 commented on this pull request.

________________________________

In Calendar/Get-CalendarDiagnosticObjectsSummary.ps1<https://github.com/microsoft/CSS-Exchange/pull/1979#discussion_r1532089300>:

> @@ -257,6 +256,66 @@ function GetMailbox {
     }
 }

+<#
+.SYNOPSIS
+Checks if a set of Calendar Logs is from the Organizer.
+#>
+function SetIsOrganizer {
+    param(
+        $CalLogs
+    )
+    $IsOrganizer = $false
+
+    foreach ($CalLog in $CalLogs) {
+        if ($CalendarItemTypes.($CalLog.ItemClass) -eq "IpmAppointment" -and
+            $CalLog.ExternalSharingMasterId -eq "NotFound" -and
+            ($CalLog.ResponseType -eq "1" -or $CalLogs.ResponseType -eq "Organizer")) {
+            $IsOrganizer = "True"


This should be a boolean of $true, not a string if we are originally setting $IsOrganizer = $false

________________________________

In Calendar/Get-CalendarDiagnosticObjectsSummary.ps1<https://github.com/microsoft/CSS-Exchange/pull/1979#discussion_r1532089927>:

> +    }
+    Write-Verbose "IsOrganizer: [$IsOrganizer]"
+    return $IsOrganizer
+}
+
+function SetIsRoom {
+    param(
+        $CalLogs
+    )
+    $IsRoom = $false
+    # Simple logic is if RBA is running on the MB, it is a Room MB, otherwise it is not.
+    foreach ($CalLog in $CalLogs) {
+        if ($CalendarItemTypes.($CalLog.ItemClass) -eq "IpmAppointment" -and
+            $CalLog.ExternalSharingMasterId -eq "NotFound" -and
+            $CalLog.Client -eq "ResourceBookingAssistant" ) {
+            $IsRoom = "True"


Same thing here. Let's be consistent, either a string or a boolean

________________________________

In Calendar/Get-CalendarDiagnosticObjectsSummary.ps1<https://github.com/microsoft/CSS-Exchange/pull/1979#discussion_r1532090444>:

> +    }
+    return $IsRoom
+}
+
+function SetIsRecurring {
+    param(
+        $CalLogs
+    )
+    Write-Host -ForegroundColor Yellow "Looking for signs of a recurring meeting."
+    $IsRecurring = $false
+    # See if this is a recurring meeting
+    foreach ($CalLog in $CalLogs) {
+        if ($CalendarItemTypes.($CalLog.ItemClass) -eq "IpmAppointment" -and
+            $CalLog.ExternalSharingMasterId -eq "NotFound" -and
+            $CalLog.CalendarItemType.ToString() -eq "RecurringMaster" ) {
+            $IsRecurring = "True"


Same thing here. Let's be consistent, either a string or a boolean

________________________________

In Calendar/Get-CalendarDiagnosticObjectsSummary.ps1<https://github.com/microsoft/CSS-Exchange/pull/1979#discussion_r1532102308>:

>          if ($cNameMatch[-1] -match "-[\w* -.]*") {
-            Write-Verbose "BetterThanNothingCNConversion: Returning : [$($cNameMatch[-1])]"
-            return $cNameMatch.split('-')[-1]
+            Write-Verbose "BetterThanNothingCNConversion: Matched : [$($cNameMatch[-1])]"


Wouldn't it be better if you just do the last X number of the CN? If we don't want it to be so long, that might be an idea.

—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/CSS-Exchange/pull/1979#pullrequestreview-1948896497>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABPSHT2LPSHKJHFDN6HOQFLYZGHO5AVCNFSM6AAAAABDALMRT2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNBYHA4TMNBZG4>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
